packages/react-button/src/utils/useToggleable.ts

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useControllableState } from '@fluentui/react-utilities';
+import type { ButtonState } from '../components/Button/Button.types';
+import type { ToggleButtonProps, ToggleButtonState } from '../components/ToggleButton/ToggleButton.types';
+
+export function useToggleable<
+  TToggleButtonProps extends ToggleButtonProps,
+  TButtonState extends ButtonState,
+  TToggleButtonState extends ToggleButtonState & TButtonState
```

Comment:
Are the use of these generics necessary ? And further, is the tight couplig to state necessary ?

It seems like `useToggleable` could just define its own type that it can work with as input. This could be done with `Pick` on `ButtonState`.

Looking at the usages, it seems like the generics are simply to result in the `TToggleButtonState`. In this case, would it just be possible to return something like `ToggleState` ?

some pseudocode below to illustrate what I'm thinking, the hook could return the state needed to make something toggleable, but it would be up to the caller to use this state appropriately

```tsx
const { onClick, role, checked } = useToggleable({ checked, defaultChecked, disabled, disabledFocusable });

const state = {
  onClick: mergeCallback(props.onClick, onClick),
  role: props.role ?? role
}
```

It makes the this hook a pure encapsulation of what a button needs to be toggleable, but leaves flexibility on how to use that state.

Reply:
I've changed this, hopefully it's better now.